### PR TITLE
refactor: integrate @doist/cli-core 0.4.0 + 0.5.0

### PR DESCRIPTION
**Pull request by scottlovegrove**
## Summary

Pulls in two cli-core releases and adopts the new shared helpers across todoist-cli.

### From [cli-core#6](https://github.com/Doist/cli-core/pull/6) (0.4.0)
- Bump \`@doist/cli-core\` 0.3.0 → 0.4.0.
- \`lib/options.ts\`: \`ViewOptions\` now extends cli-core's shared \`ViewOptions\` so json/ndjson are sourced from the canonical type.
- Adopt \`printEmpty\` in every command whose \`--json\` contract is a bare array: \`apps list\`, \`user list\`, \`hc search\`. Empty branch is now consolidated ahead of the format gate; for \`apps list\` this also fixes the dangling-newline edge case on empty \`--ndjson\`.
- Wire \`describeEmptyMachineOutput\` from \`@doist/cli-core/testing\` into \`apps\`, \`user\`, and \`hc\` test suites so the empty-output contract (\`--json\` → \`'[]\n'\`, \`--ndjson\` → \`''\`, human → message) is enforced uniformly.
- Paginated commands (\`{results, nextCursor}\` wrapper) intentionally stay on the existing format. Switching them would be a wire-format break — separate decision.

### From [cli-core#7](https://github.com/Doist/cli-core/pull/7) (0.5.0)
- Bump \`@doist/cli-core\` 0.4.0 → 0.5.0.
- \`lib/global-args.ts\`: delegates the canonical shape to cli-core's \`parseGlobalArgs\` + \`createGlobalArgsStore<TdGlobalArgs>\`. Local module now layers todoist's \`--user <ref>\` and \`--raw\` over the shared shape.
- \`isAccessible\` built via \`createAccessibleGate({ envVar: 'TD_ACCESSIBLE' })\`.
- \`shouldDisableSpinner\` built via \`createSpinnerGate({ envVar: 'TD_SPINNER' })\`.
- \`stripUserFlag\`, \`getRequestedUserRef\`, \`isRawMode\` stay local.

### New end-user surface
- \`td user list --ndjson\` and \`td hc search <q> --ndjson\` now work (one JSON value per line, no trailing newline).

### Behavior change to flag
- \`--progress-jsonl <path>\` (space-separated) is no longer parsed as a path by the pre-commander parser. Only the bare flag (\`--progress-jsonl\`, stderr) and \`--progress-jsonl=<path>\` are recognised. cli-core 0.5.0 deliberately drops the space form cross-CLI because it silently consumed the next positional (\`td task add --progress-jsonl \"Buy milk\"\` would treat \`Buy milk\` as a file path). Migrate to \`--progress-jsonl=<path>\`.

## Test plan
- [x] \`npm test\` — 1582 tests pass across 58 files (3 new contract suites for \`apps\`/\`user\`/\`hc\`).
- [x] \`npm run check\` — oxlint + oxfmt clean.
- [x] \`npm run type-check\` — clean.
- [ ] Smoke (machine output): \`td hc search \"qzqzqznoresults\" --json\` → \`[]\`; \`td hc search \"qzqzqznoresults\" --ndjson | wc -c\` → \`0\`; \`td user list --ndjson\` → one JSON per line.
- [ ] Smoke (gates): \`TD_SPINNER=false td today\`, \`td --no-spinner today\`, \`td -vq today\` → spinner off; \`td --accessible today\`, \`TD_ACCESSIBLE=1 td today\` → accessible markers.
- [ ] Smoke (progress-jsonl change): \`td --progress-jsonl=/tmp/p.jsonl today\` writes to file; \`td --progress-jsonl /tmp/p.jsonl today 2>/tmp/p2.jsonl\` now writes to \`/tmp/p2.jsonl\` (stderr) — \`/tmp/p.jsonl\` stays a positional.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/global-args.ts [hunk 1]
```diff
@@ -1,171 +1,108 @@
...
-function getGlobalArgs(): GlobalArgs {
-    if (!cached) {
-        cached = parseGlobalArgs()
-    }
-    return cached
+/**
+ * Parse well-known global flags from an argv array. Pure — pass an explicit
+ * array for testing, or omit to read `process.argv.slice(2)`.
+ *
+ * `--progress-jsonl` accepts only the bare form (output to stderr) and
```
Comment (doistbot) [inline src/lib/global-args.ts]: [P1] This only changes the pre-Commander parser. The top-level option is still declared as `--progress-jsonl [path]` in `src/index.ts`, so `td --progress-jsonl /tmp/out today` will still be accepted by Commander, but `parseGlobalArgs()` now resolves `progressJsonl` to `true` and the tracker writes to stderr. That silently ignores the user-supplied file path. To actually drop the space-separated form, the Commander option/help needs to stop using `[path]` (or argv needs to be normalized before Commander parses).

File: src/commands/hc/index.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ Notes:
         .option('--locale <locale>', LOCALE_OPTION_DESCRIPTION)
         .option('--limit <n>', 'Number of results to return (default: 10, max: 25)')
         .option('--json', 'Output as JSON')
+        .option('--ndjson', 'Output as newline-delimited JSON')
```
Comment (doistbot) [inline src/commands/hc/index.ts:55]: [P2] This adds new CLI surface, but the required agent skill docs were not updated with it. `src/lib/skills/content.ts` and the generated `skills/todoist-cli/SKILL.md` still omit `td hc search --ndjson` / `td user list --ndjson`, so installed agent skills will keep advertising the old surface. Please update and sync those files alongside these flag additions.

File: src/commands/hc/search.ts [hunk 3]
```diff
@@ -19,13 +21,18 @@ export async function searchHelpCenterArticles(
...
+
     if (options.json) {
         console.log(JSON.stringify(results, null, 2))
         return
     }
 
-    if (results.length === 0) {
-        console.log(`No Help Center articles found for "${trimmedQuery}".`)
+    if (options.ndjson) {
+        console.log(results.map((r) => JSON.stringify(r)).join('\n'))
```
Comment (doistbot) [inline src/commands/hc/search.ts]: [P2] Missed reuse: use `formatNdjson(results)` (available from `src/lib/output.ts` or directly from `@doist/cli-core`) instead of manually stringifying and joining the array.

File: src/commands/user/list.ts [hunk 4]
```diff
@@ -1,38 +1,44 @@
...
-            ),
-        )
+        console.log(JSON.stringify(projected, null, 2))
         return
     }
 
-    if (users.length === 0) {
-        console.log(chalk.dim('No stored Todoist accounts. Run `td auth login` to add one.'))
+    if (options.ndjson) {
+        console.log(projected.map((u) => JSON.stringify(u)).join('\n'))
```
Comment (doistbot) [inline src/commands/user/list.ts]: [P2] Missed reuse: use `formatNdjson(projected)` instead of manually stringifying and joining the array. The same applies to reusing `formatJson` for the `--json` branch on line 36.

File: src/commands/user/list.ts [hunk 5]
```diff
@@ -1,38 +1,44 @@
...
 
+    if (users.length === 0) {
+        printEmpty({
+            options,
+            message: chalk.dim('No stored Todoist accounts. Run `td auth login` to add one.'),
+        })
+        return
+    }
+
+    const projected = users.map((u) => ({
```
Comment (doistbot) [inline src/commands/user/list.ts]: [P3] `projected` is built eagerly for every `td user list` invocation, but the default human-readable branch below never uses it and then iterates `users` again. Move this mapping inside the `json`/`ndjson` branches (or render the human path from `projected`) so the common non-machine path avoids the extra allocation and full second pass.

File: src/commands/user/user.test.ts [hunk 6]
```diff
@@ -54,11 +55,14 @@ describe('user command', () => {
...
-
-            expect(consoleSpy.mock.calls.flat().join('\n')).toContain('No stored Todoist accounts')
+        describeEmptyMachineOutput('empty machine output contract', {
+            setup: () => {
+                mockListStoredUsers.mockResolvedValue([])
+            },
+            run: async (extraArgs) => {
+                await createProgram().parseAsync(['node', 'td', 'user', 'list', ...extraArgs])
+            },
+            humanMessage: /No stored Todoist accounts/,
```
Comment (doistbot) [inline src/commands/user/user.test.ts:65]: [P2] The `--ndjson` format is a new end-user feature introduced in this PR. While `describeEmptyMachineOutput` verifies its empty state, there is no test verifying that `td user list --ndjson` correctly emits newline-delimited JSON when stored accounts exist. Please add a test for the populated scenario to replace the manual smoke test.

File: src/commands/hc/hc.test.ts [hunk 7]
```diff
@@ -83,6 +84,16 @@ describe('hc command', () => {
...
 
+    describeEmptyMachineOutput('hc search empty machine output contract', {
+        setup: () => {
+            fetchSpy.mockResolvedValue(createJsonResponse({ results: [] }))
+        },
+        run: async (extraArgs) => {
+            await createProgram().parseAsync(['node', 'td', 'hc', 'search', 'nope', ...extraArgs])
+        },
+        humanMessage: 'No Help Center articles found for "nope".',
+    })
```
Comment (doistbot) [inline src/commands/hc/hc.test.ts:95]: [P2] Similar to `user list`, please add a test verifying that `td hc search <query> --ndjson` correctly emits newline-delimited JSON when articles are found, ensuring the new format is fully covered by automated tests beyond just the empty state.

**Review by doistbot — COMMENTED** [review]
This PR successfully integrates the latest `@doist/cli-core` releases, cleanly adopting shared helpers and standardizing machine-readable outputs across the CLI. The consolidation of empty state branches and unified global arguments significantly improves the consistency of the codebase. A few refinements are needed before merging, specifically addressing a parsing discrepancy with the `--progress-jsonl` Commander option, updating the agent skill documentation for the newly added `--ndjson` flags, reusing the available JSON formatting helpers, optimizing an eager array allocation in the user list command, and expanding test coverage for populated `--ndjson` states.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-cli&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-cli%2Fpull%2F317&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Af201ede0-4b22-11f1-957f-c7e281867540%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Af201ede0-4b22-11f1-957f-c7e281867540&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/lib/global-args.ts (hunk 1 above)
Comment (scottlovegrove) [reply src/lib/global-args.ts]: Restored space-form support for `--progress-jsonl <path>` in the local parser (76b8eb8) so it stays in lockstep with commander's `[path]` declaration. The strict cli-core 0.5.0 parser handles every other canonical flag; `--progress-jsonl` is layered on top because the flag is global (parsed before any subcommand context), so the cross-CLI `td task add --progress-jsonl "Buy milk"` confusion that motivated the strictness doesn't apply here. Tests reverted to the prior expected behaviour.

File: src/commands/hc/index.ts (hunk 2 above)
Comment (scottlovegrove) [reply src/commands/hc/index.ts:55]: Updated `src/lib/skills/content.ts` to surface the new `--ndjson` flag for both `td user list` and `td hc search` and regenerated `skills/todoist-cli/SKILL.md` via `npm run sync:skill` (76b8eb8).

File: src/commands/hc/search.ts (hunk 3 above)
Comment (scottlovegrove) [reply src/commands/hc/search.ts]: Done in 76b8eb8 — both branches now use `formatJson` / `formatNdjson` from `@doist/cli-core`.

File: src/commands/user/list.ts (hunk 4 above)
Comment (scottlovegrove) [reply src/commands/user/list.ts]: Switched both branches to `formatJson` / `formatNdjson` in 76b8eb8.

File: src/commands/user/list.ts (hunk 5 above)
Comment (scottlovegrove) [reply src/commands/user/list.ts]: Hoisted into a `projectUser` helper and moved the `.map` calls inside the json/ndjson branches (76b8eb8). The human-readable path no longer pays for the projection.

File: src/commands/user/user.test.ts (hunk 6 above)
Comment (scottlovegrove) [reply src/commands/user/user.test.ts:65]: Added a populated `--ndjson` test in 76b8eb8 — asserts two stored accounts emit one JSON value per line, correct `isDefault` flags, and no trailing newline.

File: src/commands/hc/hc.test.ts (hunk 7 above)
Comment (scottlovegrove) [reply src/commands/hc/hc.test.ts:95]: Added a populated `--ndjson` test in 76b8eb8 — two articles, one JSON per line, body field stripped, no trailing newline.
